include/ck/tensor_operation/gpu/device/device_softmax.hpp

```diff
@@ -0,0 +1,210 @@
+#ifndef DEVICE_SOFTMAX_HPP
+#define DEVICE_SOFTMAX_HPP
+
+#include <iostream>
+#include <sstream>
+#include "device.hpp"
+#include "device_base.hpp"
+#include "device_reduce.hpp"
+#include "device_reduce_multiblock.hpp"
+#include "device_reduce_common.hpp"
+#include "gridwise_softmax.hpp"
+#include "gridwise_set_buffer_value.hpp"
+#include "reduction_operator.hpp"
+
+namespace ck {
+namespace tensor_operation {
+namespace device {
+
+template <typename InDataType,
+          typename AccDataType,
+          typename OutDataType,
+          typename ScalarDataType,
```

Comment:
Suggest to remove `ScalarDataType` to avoid the Device interface API's dependency on more types.  

Currently for Reduction, for simplicity,  `float` is used for scaling value (alpha, beta) type on Device interface layer and `AccDataType` is used for scaling value type on grid-wise kernel layer.  

@asroy  What's your opinion ?

Reply:
Removed `ScalarDataType`. But kept alpha/beta type as `AccDataType`. https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/284/commits/c82f538925ff70b8600c2bce15088ae2bf656e45